private/pkg/git/gittest/gittest.go

```diff
@@ -34,9 +34,33 @@ const (
 	DefaultRemote = "origin"
 )
 
-func ScaffoldGitRepository(t *testing.T) git.Repository {
+type Repository interface {
+	git.Repository
+	Commit(ctx context.Context, t *testing.T, msg string, files map[string]string)
+	Checkout(ctx context.Context, t *testing.T, branch string)
+	CheckoutB(ctx context.Context, t *testing.T, branch string)
+	Tag(ctx context.Context, t *testing.T, msg string)
+}
+
+type scaffoldOptions struct {
+	withOnlyInitialCommit bool
+}
+
+type ScaffoldGitRepositoryOption func(*scaffoldOptions)
+
+func ScaffoldGitRepositoryWithOnlyInitialCommit() ScaffoldGitRepositoryOption {
+	return func(opts *scaffoldOptions) {
+		opts.withOnlyInitialCommit = true
+	}
+}
```

Comment:
I think this should be the default option. What about just receiving the repo and then passing it to another func that adds all the extra stuff, and that's invoked by only the tests that need it?

Reply:
I toyed with this, but the main thing is that the default scaffolding right now does things I don't want to expose yet, like `Push` and `AddWithX`. I could invert this to `WithSampleCommits`, but I'd still place the initial README commit there. What do you think?

Reply:
My main issue is that `WithSampleCommits` will only make sense to a bunch of func callers, not potentially all of them, and that's the reason why you wanted to remove the duplicated func, so everyone who needs a base git repo to play around can call this func.

What about having a runner in the caller side that can do the custom things you're thinking at the repo dir, while `Push` and `AddWithX` are ready to be exposed?

Reply:
I could do `runCommand`, but I wanted to prevent access to the raw dir (a.,.a., write file). 

aside: storagegit also uses this sample data to assert on modules that were read.

Reply:
While the repo doesn't have all exported methods to let the callers do their thing (ie commit, push, write files, etc), then I think it's safe to pass the raw directory to the caller. It's a testing pkg anyway.

Non blocking on this, but I really think this should be handled by each caller and not by the scaffolding itself.

Reply:
Done!